tpcfecutils.sh

```diff
@@ -0,0 +1,72 @@
+package: TpcFecUtils
+version: "%(tag_basename)s"
+tag: v0.1.0
+requires:
+  - boost
+  - "GCC-Toolchain:(?!osx)"
+  - Python
+  - ReadoutCard
+  - LLA
+build_requires:
+  - CMake
+source: https://gitlab+deploy-token-1303:ivjQdcMRX9qpxdv4RCcM@gitlab.cern.ch/alice-tpc-upgrade/alice-tpc-fec-utils.git
+incremental_recipe: |
+  make ${JOBS:+-j$JOBS} install
+  mkdir -p $INSTALLROOT/etc/modulefiles && rsync -a --delete etc/modulefiles/ $INSTALLROOT/etc/modulefiles
+---
+#!/bin/bash -ex
+
+# Handle submodule business
+pushd $SOURCEDIR
+git config submodule.submodules/gbt-hdlc.url \
+  https://gitlab+deploy-token-1305:2SyHnx1Tk8Rc8dY5AV9s@gitlab.cern.ch/alice-tpc-upgrade/gbt-hdlc-light.git
+git submodule update --init
+popd
+
+case $ARCHITECTURE in
+    osx*) [[ ! $BOOST_ROOT ]] && BOOST_ROOT=$(brew --prefix boost);;
+esac
+
+# Enforce no warning code in the PR checker
+if [[ $ALIBUILD_O2_TESTS ]]; then
+  CXXFLAGS="${CXXFLAGS} -Werror -Wno-error=deprecated-declarations"
+fi
+
+cmake $SOURCEDIR                                  \
+      -DBUILD_FOR_READOUT_CARD=CRU                \
+      -DBUILD_FOR_CRU_HDLC_CORE=CERN_ME           \
+      -DCMAKE_INSTALL_PREFIX=$INSTALLROOT         \
+      ${BOOST_REVISION:+-DBoost_ROOT=$BOOST_ROOT} \
+      -DCMAKE_EXPORT_COMPILE_COMMANDS=ON
+
+cp ${BUILDDIR}/compile_commands.json ${INSTALLROOT}
+
+# Reduce number of jobs if invoked by Jenkins
+if [ ! "X$JENKINS_HOME" = X ]; then
+  JOBS=1
+fi
+make ${JOBS+-j $JOBS} install
+
+#ModuleFile
+mkdir -p etc/modulefiles
```

Comment:
I would suggest you use the new `alibuild-generate-module` utility which you can find in `alibuild-recipe-tools`. Have a look at e.g. the `boost.sh` recipe to see how it's used.

Reply:
I will merge this, you can fix it later.